activerecord/lib/active_record/attribute_mutation_tracker.rb

```diff
@@ -31,7 +31,7 @@ def change_to_attribute(attr_name)
     end
 
     def any_changes?
-      attr_names.any? { |attr| changed?(attr) }
+      attr_names.any? { |attr| changed?(attr) } || forced_changes.any?
```

Comment:
Another way to fix this issue is to include virtual attributes in the attr_names. The changed? method already checks for forced changes